src/main/java/seedu/address/model/application/Application.java

```diff
@@ -0,0 +1,125 @@
+package seedu.address.model.application;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import seedu.address.model.tag.Tag;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Represents an Application in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Application {
+
+    // Identity fields
+    private final CompanyName companyName;
```

Comment:
Maybe the class names can be shorter, like Company, Position, Contact? It's not important but a bit less typing in the future. But more importantly, in case we need to add more things like details about a company, it's easier to expand (since it's less logical if CompanyName also has a field like detail for example).

Reply:
Thanks for the suggestion, fixed in [this commit](https://github.com/AY2223S1-CS2103-F14-3/tp/pull/53/commits/b837180dc55cce158758a1155219b27dbc927265)